### PR TITLE
Update from node-sass to dart-sass

### PR DESCRIPTION
**Pull request by deanmarcussen**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/8325

Because the pipeline won't work for me anymore on node 15...

replaces `gulp-sass` with it's newer `gulp-dart-sass` 

Would be great if someone on windows could try this branch and make sure it builds ok for them too.

Mostly we end up with formatting changes on the non minified files, but the dart compiler found an issue with the `switch.sass` file name, so fixed that as well.

/cc @Skrypt @jptissot @agriffard

### CODE REVIEW

File: src/OrchardCore.Themes/TheAdmin/Assets/scss/modules/admin/_switch.scss [hunk 1]
```diff
@@ -20,7 +20,7 @@ $custom-control-indicator-disabled-bg: #e9ecef;
 $custom-control-description-disabled-color: #868e96;
 $white: white;
 $theme-colors: (
-    "$success": $success,
+    "success": $success,
```
Comment (deanmarcussen) [inline src/OrchardCore.Themes/TheAdmin/Assets/scss/modules/admin/_switch.scss:23]: @Skrypt this was the change I had to make to get it to compile.

when I look at the css we were generating, there were instances of
`.alert-$success` in `TheAdmin.css`

Those gone now

Comment (Skrypt) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/modules/admin/_switch.scss:23]: These switch are not maintained anymore. They are missing their javascript, I think it should probably be removed simply.

File: src/OrchardCore.Themes/TheComingSoonTheme/wwwroot/gulpfile.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const header = require("gulp-header");
 const merge = require("merge-stream");
 const plumber = require("gulp-plumber");
 const rename = require("gulp-rename");
-const sass = require("gulp-sass");
+const sass = require("gulp-dart-sass");
```
Comment (Skrypt) [inline src/OrchardCore.Themes/TheComingSoonTheme/wwwroot/gulpfile.js:13]: These might get overwritten when @agriffard will update the theme.

File: src/OrchardCore.Themes/TheComingSoonTheme/wwwroot/package.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
         "gulp-header": "2.0.9",
         "gulp-plumber": "^1.2.1",
         "gulp-rename": "2.0.0",
-        "gulp-sass": "4.1.0",
+        "gulp-dart-sass": "1.0.2",
```
Comment (Skrypt) [inline src/OrchardCore.Themes/TheComingSoonTheme/wwwroot/package.json:45]: Same

**Comment by Skrypt** [comment]
Still works with node 14.15.4

**Comment by Skrypt** [comment]
Dark mode alert-success needs some adjustments. Strange one.

File: src/OrchardCore.Themes/TheComingSoonTheme/wwwroot/gulpfile.js (hunk 2 above)
Comment (deanmarcussen) [reply src/OrchardCore.Themes/TheComingSoonTheme/wwwroot/gulpfile.js:13]: Yeah was thinking might just pr the source to upgrade it as well. To save that issue.

**Comment by deanmarcussen** [comment]
ok reverted the theme changes, and removed building the themes from the primary gulpfile.

They really shouldn't have been there anyway, they're not part of the `assets.json` arrangement, and because they are external, are more dificult to control and manage.

To script the building of those for @agriffard I think they should just be part of the npm scripts(i.e. `npm run build-themes`), not part of the task runner.

**Review by Skrypt — APPROVED** [review]

**Pull request merged**